src/target/stm32f1.c

```diff
@@ -440,6 +444,7 @@ static bool stm32f1_option_write(target *t, uint32_t addr, uint16_t value)
 	index = (addr - FLASH_OBP_RDP) / 2;
 	if ((index < 0) || (index > 7))
 		 return false;
+
 	/* Retrieve old values */
 	for (i = 0; i < 16; i = i +4) {
```

Comment:
This feels like i is defined elsewhere and whoever wrote this doesn't know about += - please rewrite this using size_t for i and `i += 4`